### DISCUSSION TITLE
How to disable Full Refresh Modal (and just do it?)

### DISCUSSION THREAD

**Discussion by deanrad** [vercel/next.js#35886] [Help]
**Next 12.0.10**

The modal titled "About to perform a full refresh" essentially nullifies the benefits of hot reloading.
For 2 decades hot reloading was equivalent to a full refresh, the thing the dialog warns us about. If I have to use the mouse (or keyboard) to click a modal _I may as well have clicked the refresh button myself!_

Can I just disable the warning and have it reload automatically?

Please help me solve my way out of this workflow-dragging situation.

**Comment by icyJoseph** [comment]
Isn't there like a `Don't show again for session` option?

I disagree very much with your observations though. Hot reloading is not equal to full refresh. A full refresh is quite the contrary, it means that because a little thing changed, we ought to destroy the entire page, as opposed to hot swap the changed area. And for about 2 decades, iirc the first time I saw hot module reloading was in 2015, with redux reducers. 

Try with a more recent version of `Next.js` I have a feeling `12.0.10`, might have had a bug https://github.com/vercel/next.js/issues/31827 regarding the overlay.

**Reply by deanrad** [reply]
@icyJoseph 

I did find `// @refresh reset` in the docs- however it doesn't help, in my case.

I suspect it could be #31827 mistakenly raising the dialog because we use middleware, so when we upgrade next maybe we don't get it.. thanks for the tip..

You're correct, I confused my reloading models. Let's call it 'browser updating' more generally.

When Next figures out a full refresh is needed to update the browser, why ask? If it fears to reload "too much" (so timidly warns) I am no longer getting the benefit of any auto updating. If it reloads "too much", it keeps me productive. Please, reload too much when you detect it's necessary!

That 'disable for this session option', would be nice, but I don't see in my version, or in the live docs - what version adds that?

**Reply by icyJoseph** [reply]
I think `next 12.1.4` should contain that. I must say that I've also started to see that full reload modal way less. I can't even make it happen again. Please do try a newer version of Next and see if it persist, you don't have to publish your app with the newest version, but at least you'll know it was "fixed" already.

**Reply by deanrad** [reply]
Yep, as of 12.1.0 there's a disable for this session dialog, cool! Maybe the docs could mention that.. (though people will see it, and that's probably good enough).
